src/Core/src/Platform/iOS/MauiTextView.cs

```diff
@@ -161,5 +173,21 @@ void ShouldCenterVertically()
 				_ => new CGPoint(0, 0),
 			};
 		}
+
+		void UpdatePlaceholderFontSize(UIFont? value)
+		{
+			if (value != null)
+			{
+				if (_defaultPlaceholderSize == -1)
+				{
+					_defaultPlaceholderSize = _placeholderLabel.Font.PointSize;
+				}
+				_placeholderLabel.Font = _placeholderLabel.Font.WithSize(value.PointSize);
+			}
+			else if (_defaultPlaceholderSize != -1)
+			{
+				_placeholderLabel.Font = _placeholderLabel.Font.WithSize(_defaultPlaceholderSize);
+			}
+		}
```

Comment:
I don't quite understand the logic, lets walk through it, first invert the if, that results in the following:
```suggestion
		void UpdatePlaceholderFontSize(UIFont? value)
		{
			if (value is null) {
				if (_defaultPlaceholderSize != -1)
					_placeholderLabel.Font = _placeholderLabel.Font.WithSize (_defaultPlaceholderSize);
			}
			else {
				if (_defaultPlaceholderSize == -1) {
					_defaultPlaceholderSize = _placeholderLabel.Font.PointSize;
				}

				_placeholderLabel.Font = _placeholderLabel.Font.WithSize (value.PointSize);
			}
		}
```
Same behaviour, of course if we have a nested if with no else, that equals a && so we have:
```suggestion
		void UpdatePlaceholderFontSize(UIFont? value)
		{
			if (value is null && _defaultPlaceholderSize != -1) {
				_placeholderLabel.Font = _placeholderLabel.Font.WithSize (_defaultPlaceholderSize);
			}
			else {
				if (_defaultPlaceholderSize == -1) {
					_defaultPlaceholderSize = _placeholderLabel.Font.PointSize;
				}

				_placeholderLabel.Font = _placeholderLabel.Font.WithSize (value.PointSize);
			}
		}
```

We can now move to the else clause, which is the one that confuses me, looks like you want to set _defaultPlaceholderSize to the size of the _placeholderLabel.Font.PointSize only when it is -1. By the look of the code _defaultPlaceholderSize is never modified by any other value, so why not moving that if outside the else clause?

```suggestion
		void UpdatePlaceholderFontSize(UIFont? value)
		{
		    if (_defaultPlaceholderSize == -1) {
				_defaultPlaceholderSize = _placeholderLabel.Font.PointSize;
			}
			
			if (value is null && _defaultPlaceholderSize != -1) {
				_placeholderLabel.Font = _placeholderLabel.Font.WithSize (_defaultPlaceholderSize);
			}
			else {
				_placeholderLabel.Font = _placeholderLabel.Font.WithSize (value.PointSize);
			}
		}
```
No a huge fan of a read-write variable accessible by the rest of the code to hold the default value and be updated the first time a property is used. Maybe using https://learn.microsoft.com/en-us/dotnet/api/system.lazy-1?view=net-7.0 is a better approach, but I let you judge. Nevertheless the code is cleaner this way, or at least with the first iteration.

I believe you can make it a lot nicer if you use a nullable value, rather than _defaultPlaceholderSize = -1 make it a nfloat? which is much safer, we know we started with null, if not null someone changed it:
```suggestion
		void UpdatePlaceholderFontSize(UIFont? value)
		{
		    if (_defaultPlaceholderSize is null) {
				_defaultPlaceholderSize = _placeholderLabel.Font.PointSize;
			}
			
			if (value is null && _defaultPlaceholderSize is not null) {
				_placeholderLabel.Font = _placeholderLabel.Font.WithSize (_defaultPlaceholderSize);
			}
			else {
				_placeholderLabel.Font = _placeholderLabel.Font.WithSize (value.PointSize);
			}
		}
```
You can make it even smaller using a ternary operator:
```suggestion
		void UpdatePlaceholderFontSize(UIFont? value)
		{
		    if (_defaultPlaceholderSize is null) {
				_defaultPlaceholderSize = _placeholderLabel.Font.PointSize;
			}
			
			_placeholderLabel.Font = _placeholderLabel.Font.WithSize (
					value is null && _defaultPlaceholderSize is not null ? _defaultPlaceholderSize : value.PointSize);
			}
		}
```

The last suggestion seems the cleaner one. But do as you wish :)